docs/usecase.rst

```diff
@@ -37,14 +72,37 @@ compute/data/communications abstractions,
 quality of service guarantees,
 and performance optimizations.
 
+.. seealso:: :doc:`executor`
+
 Generic sequential task execution
 ---------------------------------
 
 Units of work (nodes) describe the inputs and expected outputs,
 and identify a module able to perform the operation.
 
+| 1. Receive a node of work to execute.
+| 2. Import a module implementing the operation.
+| 3. Initialize or restore checkpoint state.
+| 4. Prepare operation resources.
+|    3.1 Request or confirm locality of input data.
+|    3.2 Create data publishing resource for operation.
+|    3.3 Prepare compute environment for execution.
+|    3.4 Use operation module helpers to construct input objects.
+| 5. Provide resources and execute operation.
```

Comment:
I think here 'provide' indicates that resources are already acquired and that are available for the work to be scheduled on them. Would you agree?

Reply:
Yes. It is a detail of the executor/worker when and how resources are acquired. At this step, the resources are packaged and passed to a callable.

Reply:
See my initial comment about the meaning of resources. In that context, note that, ultimately, the initial acquisition of the HPC resources needs to be specified and triggered by the end user.

Reply:
We may not always be able to constrain the resources available to a task, but the design should allow a place for us to do so. This would be the place, for instance, where the task receives an MPI subcommunicator instead of MPI_COMM_WORLD, or may be told how many threads to spawn.